### ISSUE TITLE
mlpack train weighted DecisionTree using sample frequency

### ISSUE THREAD

**Issue by xiandong79** [mlpack/mlpack#2735]
@rcurtin @MHenderson @jeffin143   I have the same question:

in mlpack::DecisionTree, how to "Weights" associated with classes

with https://stackoverflow.com/questions/64029932/mlpack-train-weighted-decisiontree-using-sample-frequency

it is also a feature in `sklearn.tree.DecisionTreeClassifier`
https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html

**Comment by xiandong79** [comment]
according to https://github.com/mlpack/mlpack/issues/882

it seems that `arma::rowvec& weights` is applied on each **data point**, instead of each **label** like sklearn.tree.DecisionTreeClassifier
